File: Algorithms/977. Squares of a Sorted Array/php/solution.two-pointers.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function sortedSquares($nums)
    {
        $n = count($nums);
        $result = array_fill(0, $n, 0);
        $left = 0;
        $right = $n - 1;
        $lastIndex = $n - 1;
        while ($left <= $right) {
            $leftSquare = $nums[$left] ** 2;
            $rightSquare = $nums[$right] ** 2;
            if ($leftSquare < $rightSquare) {
                $result[$lastIndex] = $rightSquare;
                $right--;
            } else {
                $result[$lastIndex] = $leftSquare;
                $left++;
            }
            $lastIndex--;
        }
        return $result;
    }
}

File: Algorithms/977. Squares of a Sorted Array/golang/solution.two-pointers.go
func sortedSquares(nums []int) []int {
	var n, lastIndex, left, right int = len(nums), len(nums) - 1, 0, len(nums) - 1
	result := make([]int, n)

	for left <= right {
		var leftSquare, rightSquare int = nums[left] * nums[left], nums[right] * nums[right]

		if leftSquare < rightSquare {
			result[lastIndex] = rightSquare
			right--
		} else {
			result[lastIndex] = leftSquare
			left++
		}
		lastIndex--
	}
	return result
}